### ISSUE TITLE
Current consumption issue with DZG DxS74 power meters; consumption < 655.36 watts get counted negative

### ISSUE THREAD

**Issue by HWiese1980** [Danielhiversen/pyTibber#266]
Hey folks,

I'm not sure whom to address with this issue or where to ask for help. My landlord is not very cooperative, neither is my grid operator. Tibber themselves have been open for suggestions, understanding, listening to my pain, but haven't solved the problem themselves yet. I don't know if they will.

There is a known issue with the firmware of the DZG DxS74 power meters. For some reason power consumption values below 655.36 get read by an IR reader like Tibber Pulse as negative values and turned into a production value instead of actual consumption. The resulting value is `production = 655.36W - actual_consumption`, e.g. a consumption of 400 watts gets read as `655.36W - 400W = 255.36W production`.

I think it can only be reliably corrected if I'm not actually producing power (say, I have no solar panels on my roof or something). If I don't, I can correct the value by doing:

```python
# assuming production = max(0, 655.36 - real_consumption) ->
if production > 0.0:  # which shouldn't happen if I have no power plant of any kind
  consumption = 655.36 - production   # correct consumption
  production = 0.0  # and production values
```

If I do produce power, it's more difficult and I haven't thought about it enough to come up with an easy fix. I myself am only consuming, not producing.

I'm wondering who should implement the fix. Of course it *should* be the manufacturer of the power meter. But who's next in line if they won't? Who'd have it the easiest to provide a fix for this? Would it be most convenient to add it to this library as a switch? Or to the Tibber integration into Home Assistant? 

I have no idea. What's your thoughts on this?

**Comment by Danielhiversen** [comment]
This library should not start correcting data from the Tibber API. It should happen earlier in the pipeline.

**Comment by HWiese1980** [comment]
I agree. It should happen as early as possible. The problem is that I do not get any productive response from any of the other components earlier in the chain.
